### PR TITLE
New assembly instructions

### PR DESCRIPTION
**Pull request by OLUWAMUYIWA**
## What
Added `assertz`, `u32wrapping_madd`, and `u32wrapping_add3` to assembly instructions.
Ref: #357 

Also updated the documentation to accommodate the new insructions.

### CODE REVIEW

**Comment by OLUWAMUYIWA** [comment]
@bobbinth please check.

File: docs/src/user_docs/assembly/field_operations.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ For instructions where one or more operands can be provided as immediate paramet
 | ---------------- | ----------- | ------------- | ----------------------------- |
 | assert  <br> - *(1 cycle)*          | [a, ...]    | [...]         | If $a = 1$, removes it from the stack. <br> Fails if $a \ne 1$ |
 | assert_eq <br> - *(2 cycles)*        | [b, a, ...] | [...]         | If $a = b$, removes them from the stack. <br> Fails if $a \ne b$ |
+| assertz <br> - *(2 cycles)*       | [ a, ...] | [...]              | if $a = 0$, removes it from the stack, <br> Fails if $a \ne b$ |
```
Comment (bobbinth) [inline docs/src/user_docs/assembly/field_operations.md:14]: minor nit: I'd probably put `assertz` above `assert_eq` in the table.

File: assembly/src/parsers/field_ops.rs [hunk 2]
```diff
@@ -30,6 +30,19 @@ pub(super) fn parse_assert_eq(
     Ok(())
 }
 
+/// Appends EQZ ASSERT operation sequence to the span block.
+pub(super) fn parse_assert_eqz(
```
Comment (bobbinth) [inline assembly/src/parsers/field_ops.rs:34]: minor nit: I'd probably name the function `parse_assertz` to keep it consistent with the instruction name.

File: assembly/src/parsers/u32_ops.rs [hunk 3]
```diff
@@ -1058,3 +1104,42 @@ fn handle_division(
...
+mod tests {
+    use vm_core::Operation;
+
+    use crate::parsers::u32_ops::U32OpMode;
+    use crate::{
+        parsers::u32_ops::{parse_u32wrapping_add3, parse_u32wrapping_madd},
+        tokens::Token,
+        AssemblyError,
+    };
+    #[test]
```
Comment (bobbinth) [inline assembly/src/parsers/u32_ops.rs:1118]: very minor nit: I'd add a blank line above `#[test]`.

**Review by bobbinth — CHANGES_REQUESTED** [review]
Looks great! Thank you! I left a couple of very minor nit comments inline.

**Comment by OLUWAMUYIWA** [comment]
@bobbinth, I made the changes you requested

**Review by bobbinth — APPROVED** [review]
All looks good! Thank you again!

**Pull request merged**